agent/api/task_routes.py

```
@@ -150,6 +150,37 @@ async def get_orphaned_tasks(
         task_service = TaskService(session, active_env=active_env)
         return task_service.get_unretried_orphaned_tasks()
 
+    @router.get("/tasks/failed/groups/recent", response_model=List[FailureGroupSummary])
+    async def get_recent_failure_groups(
+        hours: Optional[int] = Query(default=None, ge=1, le=168, description="Lookback window in hours (defaults to configured value)"),
+        limit: int = 50,
+        include_retried: bool = False,
+        session: Session = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)